packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -1208,6 +1208,47 @@ class DecoratedBoxTransition extends AnimatedWidget {
 ///
 /// {@animation 300 378 https://flutter.github.io/assets-for-api-docs/assets/widgets/align_transition.mp4}
 ///
+/// {@tool dartpad --template=stateful_widget_material_ticker}
+/// The following code implements the [AlignTransition] as seen in the video
+/// above:
+///
+/// ```dart
+/// late final AnimationController _controller = AnimationController(
```

Comment:
This a bit subtle, typically late final variables like this are initialized in initState().  This approach may work equally well, however it would be helpful to include a comment like:
// Using `late final` for [lazy initialization](https://dart.dev/null-safety/understanding-null-safety#lazy-initialization).